### PR TITLE
fix(pom): fixed the java version in kc module pom.xml

### PR DESCRIPTION
**Pull request by smrutis1**
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

Issue: 

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Review by heliocastro — APPROVED** [review]
Ok

File: keycloak/pom.xml [hunk 1]
```diff
@@ -36,8 +36,8 @@
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>3.5.1</version>
                 <configuration>
-                    <source>17</source>
-                    <target>17</target>
+                    <source>21</source>
+                    <target>21</target>
```
Comment (heliocastro) [inline keycloak/pom.xml:40]: Remove, let super pom handle it

File: keycloak/user-storage-provider/pom.xml [hunk 2]
```diff
@@ -20,8 +20,8 @@
 
     <properties>
         <version.compiler.maven.plugin>3.5.1</version.compiler.maven.plugin>
```
Comment (heliocastro) [inline keycloak/user-storage-provider/pom.xml:22]: Remove this line as well, versions should be set on super pom

File: keycloak/user-storage-provider/pom.xml [hunk 3]
```diff
@@ -20,8 +20,8 @@
...
-        <maven.compiler.source>17</maven.compiler.source>
-        <maven.compiler.target>17</maven.compiler.target>
+        <maven.compiler.source>21</maven.compiler.source>
+        <maven.compiler.target>21</maven.compiler.target>
```
Comment (heliocastro) [inline keycloak/user-storage-provider/pom.xml:24]: Just remove it, let super pom handles

File: keycloak/user-storage-provider/pom.xml [hunk 4]
```diff
@@ -20,8 +20,8 @@
...
         <version.hibernate.javax.persistence>1.0.0.Final</version.hibernate.javax.persistence>
         <arquillian-graphene.version>2.5.4</arquillian-graphene.version>
         <version.jee.jaxb.api>2.3.1</version.jee.jaxb.api>
```
Comment (heliocastro) [inline keycloak/user-storage-provider/pom.xml:27]: Add this on super pom

**Review by heliocastro — CHANGES_REQUESTED** [review]

File: keycloak/user-storage-provider/pom.xml (hunk 4 above)
Comment (smrutis1) [reply keycloak/user-storage-provider/pom.xml:27]: @heliocastro Do you mean all the above properties? or any specific.

Comment (smrutis1) [reply keycloak/user-storage-provider/pom.xml]: Moved it.

File: keycloak/user-storage-provider/pom.xml (hunk 2 above)
Comment (smrutis1) [reply keycloak/user-storage-provider/pom.xml]: Done

**Pull request merged**